social.md

```diff
@@ -11,134 +11,273 @@ support:
 
-> `<stormpath.web.social.[providerId].uri>`
+The Client API signing key can be retrieved by using the Stormpath API endpoint specified in the `signingApiKey` of the Application's `webConfig`:
 
-Where `providerId` is the property that is found in the `provider` object of
-the Stormpath directory, e.g. `google` or `facebook`.
+```
+"webConfig": {
+	...
+    "signingApiKey": {
+        "href": "https://<Stormpath API Host>/v1/apiKeys/<API Key ID>"
+    },
+    "status": "ENABLED",
+	...
+}
+```
 
-When the callback URL is requested with a GET request, the user is being
-redirected back to the server after authentication with the provider.  The
-callback handler must complete the following tasks:
+Retrieving this URL returns an api key response that includes the API Key Secret used to sign the `jwtResponse`:
 
-  * Parse the callback data from the provider to get the authorization code
-  * Perform the authorization code exchange, if needed. 
-  * Retrieve or create the account, by posting the provider data to Stormpath
-  * Create the OAuth2 token cookies for the user
-  * Redirect the user to `stormpath.web.login.nextUri`
+```
+{
+    "account": {
+        "href": "<account href>"
+    },
+    "description": "<description>",
+    "href": "<apiKeys href>",
+    "id": "<API Key ID>",
+    "name": "<name>",
+    "secret": "<API Key Secret>",
+    "status": "ENABLED",
+    "tenant": {
+        "href": "<tenant href>"
+    }
+}
+```
 
-If any of these tasks fail, an error should be immediately rendered to the user
-and the next task should not be attempted.
+Once verified, the `jwtResponse` should be processed in the usual way by the framework integration for an AUTHENTICATED response as indicated [here](/login.md#-post-response-handling).
 
-The error should be displayed on the login form in the same way other login
-errors are shown.
+## Look and Feel
 
-Recommendation: the callback endpoints should not handle access tokens in the query string because applications that log URLs will inadvertently log access tokens, which presents a security risk. 
+All of the integrations, regardless of language, should have a consistent look and feel for the rendered buttons for social providers.
 
-## Implementing Popup-Based Workflows
+**Recommendation:**
 
-Single-page applications will need to query the server to determine which social
-providers are configured, and which fields to render for the login or
-registration views.  In either context, the front-end application should make
-a GET request to the login or registration endpoint, to fetch the view model
-as JSON.  The view model will contain the necessary information.  See [login][]
-and [registration][] pages for the view model definitions.
+The [Social Buttons for Bootstrap](https://lipis.github.io/bootstrap-social/) has a very clean consistent look and feel and has been adopted into the Java SDK. Cons for this approach are a dependency on Bootstrap.
```

Comment:
As much as I ❤️ Social Buttons for Bootstrap, I am not in favor of taking a dependency on Bootstrap for all our integrations. Bootstrap is heavy. I'd prefer an approach where we didn't imply any styling at all but left that up to the implementer. SBB could still be one of the recommended/suggested approaches.